quickwit/quickwit-indexing/src/source/file_source.rs

```diff
@@ -71,48 +64,37 @@ impl Source for FileSource {
         doc_processor_mailbox: &Mailbox<DocProcessor>,
         ctx: &SourceContext,
     ) -> Result<Duration, ActorExitStatus> {
-        // We collect batches of documents before sending them to the indexer.
-        let limit_num_bytes = self.counters.previous_offset + BATCH_NUM_BYTES_LIMIT;
-        let mut reached_eof = false;
+        let limit_num_bytes = self.counters.offset + BATCH_NUM_BYTES_LIMIT;
+        let mut new_offset = self.counters.offset;
         let mut batch_builder = BatchBuilder::new(SourceType::File);
-
-        while self.counters.current_offset < limit_num_bytes {
-            let mut doc_line = String::new();
-            // guard the zone in case of slow read, such as reading from someone
-            // typing to stdin
-            let num_bytes = ctx
-                .protect_future(self.reader.read_line(&mut doc_line))
-                .await
-                .map_err(anyhow::Error::from)?;
-            if num_bytes == 0 {
-                reached_eof = true;
-                break;
+        let is_eof = loop {
```

Comment:
I tend to prefer `while loops` because they are easier to read proof.
(answering the question, can the loop go on infinitely is much simpler)